packages/deploy/utils/lazyMinting.ts

```diff
@@ -0,0 +1,357 @@
+  sandContract: Contract,
+  orderValidatorContract: Contract,
+  catalystTier: bigint,
+  amount: bigint,
+  maker: Signer,
+  taker: Signer
+) => {
+  const makerAsset = await AssetERC1155(
+    catalystContract,
+    Number(catalystTier),
+    Number(amount)
+  );
+  const takerAsset = await AssetERC20(sandContract, catalystPrice);
+
+  const randomSalt = getRandomInt(1000000);
+
+  const orderLeft = await OrderDefault(
+    maker,
+    makerAsset,
+    ZeroAddress,
+    takerAsset,
+    randomSalt,
+    0,
+    0
+  );
+
+  const orderRight = await OrderDefault(
+    taker,
+    takerAsset,
+    ZeroAddress,
+    makerAsset,
+    randomSalt,
+    0,
+    0
+  );
+
+  const makerSig = await signOrder(orderLeft, maker, orderValidatorContract);
+  const takerSig = await signOrder(orderRight, taker, orderValidatorContract);
+
+  const matchedOrder = [
+    {
+      orderLeft,
+      signatureLeft: makerSig,
+      orderRight,
+      signatureRight: takerSig,
+    },
+  ];
+
+  return matchedOrder;
+};
+
+export const createLazyMintSignature = async (
+  data: LazyMintData,
+  AssetCreateContract: Contract,
+  network: Network
+) => {
+  const {
+    caller,
+    tier,
+    amount,
+    unitPrice,
+    paymentToken,
+    metadataHash,
+    maxSupply,
+    creator,
+  } = data;
+  const nonce = await AssetCreateContract.signatureNonces(caller);
+
+  const backendAuthWallet = new ethers.Wallet(
+    '0x4242424242424242424242424242424242424242424242424242424242424242'
+  );
+
+  const sigData = {
+    types: {
+      LazyMint: [
+        {name: 'caller', type: 'address'},
+        {name: 'creator', type: 'address'},
+        {name: 'nonce', type: 'uint16'},
+        {name: 'tier', type: 'uint8'},
+        {name: 'amount', type: 'uint256'},
+        {name: 'unitPrice', type: 'uint256'},
+        {name: 'paymentToken', type: 'address'},
+        {name: 'metadataHash', type: 'string'},
+        {name: 'maxSupply', type: 'uint256'},
+      ],
+    },
+    domain: {
+      name: 'Sandbox Asset Create',
+      version: '1.0',
+      chainId: network.config.chainId,
+      verifyingContract: await AssetCreateContract.getAddress(),
+    },
+    message: {
+      caller,
+      creator,
+      nonce,
+      tier,
+      amount,
+      unitPrice,
+      paymentToken,
+      metadataHash,
+      maxSupply,
+    },
+  };
+
+  const signature = await backendAuthWallet.signTypedData(
+    sigData.domain,
+    sigData.types,
+    sigData.message
+  );
+  return signature;
+};
+
+export const createMultipleLazyMintSignature = async (
+  data: LazyMintBatchData,
+  AssetCreateContract: Contract,
+  network: Network
+) => {
+  const {
+    caller,
+    creators,
+    tiers,
+    amounts,
+    unitPrices,
+    paymentTokens,
+    metadataHashes,
+    maxSupplies,
+  } = data;
+  const nonce = await AssetCreateContract.signatureNonces(caller);
+  const backendAuthWallet = new ethers.Wallet(
+    '0x4242424242424242424242424242424242424242424242424242424242424242'
```

Comment:
Wonder whether it is better to grab this address from hardhat config rather than hardcode here?

Reply:
I've tried to do it like:

```js
  const {backendAuthWallet} = await getNamedAccounts();
  const backendAuthWalletSigner = await ethers.getSigner(backendAuthWallet);
```

But it throws an error:

` InvalidInputError: unknown account 0x17c5185167401ed00cf5f5b2fc97d9bbfdb7d025`

Thats also how its done in Exchange tests so I followed that one.